examples/step-10/step-10.cc

```diff
@@ -64,14 +65,19 @@ namespace Step10
   // of the library). Rather, we just pack the functionality into separate
   // functions. We make these functions templates on the number of space
   // dimensions to conform to usual practice when using deal.II, although we
-  // will only use them for two space dimensions.
+  // will only use them for two space dimensions and throw an exception when
+  // attempted to use for any other spatial dimension
   //
   // The first of these functions just generates a triangulation of a circle
   // (hyperball) and outputs the $Q_p$ mapping of its cells for different values
   // of <code>p</code>. Then, we refine the grid once and do so again.
   template <int dim>
   void gnuplot_output()
   {
+    Assert(
+      dim == 2,
+      ExcMessage(
+        "This tutorial is currently implemented only for two spatial dimensions "));
```

Comment:
Doesn't this particular function also make sense for 3d?

Reply:
Yes, it does. However, this particular tutorial introduction claims 
>We make these functions templates on the number of space dimensions to conform to usual practice when using deal.II, although we will only use them for two space dimensions.

And, all the results are provided only for 2 spatial dimensions (circle). And in fact, this is about calculating the value of pi.

I made this tutorial exactly replicating the line of explanation given in tutorial 5 about how to use Assertions.

> 
> Assert(dim == 2, ExcInternalError());
> 
> ExcInternalError is a globally defined exception, which may be thrown whenever something is terribly wrong. Usually, one would like to use more specific exceptions, and particular in this case one would of course try to do something else if dim is not equal to two, e.g. create a grid using library functions. Aborting a program is usually not a good idea and assertions should really only be used for exceptional cases which should not occur, but might due to stupidity of the programmer, user, or someone else. The situation above is not a very clever use of Assert, but again: this is a tutorial and it might be worth to show what not to do, after all.
> 
> So if we got past the assertion, we know that dim==2, and we can now actually read the grid. It is in UCD (unstructured cell data) format (though the convention is to use the suffix inp for UCD files):
> grid_in.read_ucd(input_file);

Reply:
@masterleinad Are you happy with this? Shall we mark this as resolved?

Reply:
Since there was no feedback, I will remove this assertion from the commit.